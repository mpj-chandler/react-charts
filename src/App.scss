*, *::before, *::after {
  margin: 0;
  padding: 0;
  font-family: Arial, sans-serif;
  font-size: calc(12px + 0.5vw);
}

.App {
  width: 100vw;
  height: 100vh;
  display: grid;
  grid-template-areas: 'title title' 'top-left top-right' 'bottom-left bottom-right';
  grid-template-rows: 10vh 1fr 1fr;
  grid-template-columns: 1fr 1fr;

  &__Title {
    grid-area: title;
    text-align: center;
    align-items: center;
  }

  &__ChartContainer {
    text-align: center;
    grid-area: top-left;
    border: 1px solid grey;
    border-radius: 0.25rem;
    align-content: center;
  }
}
