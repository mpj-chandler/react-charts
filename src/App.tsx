import React from 'react';
import styles from './App.scss';
import BarChart, {BarChartProps} from './components/BarChart/BarChart';

interface AppProps {

}

const App: React.FC<AppProps> = (props) => {
    const barChartProps: BarChartProps = {
        title: 'My first bar chart',
        data: [
            {
                seriesName: 'A',
                points: [
                    {
                        x: 0,
                        y: 10
                    },
                    {
                        x: 1,
                        y: 20
                    },
                    {
                        x: 2,
                        y: 30
                    }
                ]
            }
        ],
        padding: { left: 10, right: 10, top: 10, bottom: 10 },
        xAxisConfig: {
            zeroIntercept: true,
            margin: 10
        },
        yAxisConfig: {
            zeroIntercept: true,
            margin: 10
        }
    }
    return (
        <div className={styles.App} {...props}>
            <span className={styles.App__Title}>React Charts</span>
            <div className={styles.App__ChartContainer}>
                <BarChart {...barChartProps}/>
            </div>
        </div>
    );
};

export default App;
