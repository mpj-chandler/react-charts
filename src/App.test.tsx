import React from 'react';
import TestRenderer from 'react-test-renderer';
import App from './App';

jest.mock('./components/BarChart/BarChart.tsx');

describe('App', () => {
    it('it renders consistently', () => {
        const component = TestRenderer.create(<App/>);

        expect(component.toJSON()).toMatchSnapshot();
    })
})
