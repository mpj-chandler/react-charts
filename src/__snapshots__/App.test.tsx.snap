// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App it renders consistently 1`] = `
<div
  className="App"
>
  <span
    className="App__Title"
  >
    React Charts
  </span>
  <div
    className="App__ChartContainer"
  >
    <div
      data={
        Array [
          Object {
            "points": Array [
              Object {
                "x": 0,
                "y": 10,
              },
              Object {
                "x": 1,
                "y": 20,
              },
              Object {
                "x": 2,
                "y": 30,
              },
            ],
            "seriesName": "A",
            "type": Object {
              "x": "NON_NULL_NUMERIC",
              "y": "NON_NULL_NUMERIC",
            },
          },
        ]
      }
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      title="My first bar chart"
      xAxisConfig={
        Object {
          "margin": 10,
          "tickLength": 2,
          "tickPlacement": 1,
          "zeroIntercept": true,
        }
      }
      yAxisConfig={
        Object {
          "margin": 10,
          "tickLength": 2,
          "tickPlacement": 0,
          "zeroIntercept": true,
        }
      }
    >
      Mock BarChart 
    </div>
  </div>
  <div
    className="App__SecondChartContainer"
  >
    <div
      data={
        Array [
          Object {
            "points": Array [
              Object {
                "x": 0,
                "y": -10,
              },
              Object {
                "x": 1,
                "y": 20,
              },
              Object {
                "x": 2,
                "y": 30,
              },
            ],
            "seriesName": "A",
            "type": Object {
              "x": "NON_NULL_NUMERIC",
              "y": "NON_NULL_NUMERIC",
            },
          },
        ]
      }
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      title="My second bar chart"
      xAxisConfig={
        Object {
          "margin": 10,
          "tickLength": 2,
          "tickPlacement": 1,
          "zeroIntercept": true,
        }
      }
      yAxisConfig={
        Object {
          "margin": 10,
          "tickLength": 2,
          "tickPlacement": 0,
          "zeroIntercept": true,
        }
      }
    >
      Mock BarChart 
    </div>
  </div>
  <div
    className="App__ThirdChartContainer"
  >
    <div
      data={
        Array [
          Object {
            "points": Array [
              Object {
                "x": 0,
                "y": -10,
              },
              Object {
                "x": 1,
                "y": -20,
              },
              Object {
                "x": 2,
                "y": -30,
              },
            ],
            "seriesName": "A",
            "type": Object {
              "x": "NON_NULL_NUMERIC",
              "y": "NON_NULL_NUMERIC",
            },
          },
          Object {
            "points": Array [
              Object {
                "x": 0,
                "y": -30,
              },
              Object {
                "x": 1,
                "y": 20,
              },
              Object {
                "x": 2,
                "y": 60,
              },
            ],
            "seriesName": "B",
            "type": Object {
              "x": "NON_NULL_NUMERIC",
              "y": "NON_NULL_NUMERIC",
            },
          },
        ]
      }
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      title="My third bar chart"
      xAxisConfig={
        Object {
          "margin": 10,
          "tickLength": 2,
          "tickPlacement": 1,
          "zeroIntercept": true,
        }
      }
      yAxisConfig={
        Object {
          "margin": 10,
          "tickLength": 2,
          "tickPlacement": 0,
          "zeroIntercept": true,
        }
      }
    >
      Mock BarChart 
    </div>
  </div>
  <div
    className="App__FourthChartContainer"
  >
    <div
      data={
        Array [
          Object {
            "points": Array [
              Object {
                "x": "Anything",
                "y": -10,
              },
              Object {
                "x": "Belongs",
                "y": 20,
              },
              Object {
                "x": "Conservatory",
                "y": -30,
              },
            ],
            "seriesName": "A",
            "type": Object {
              "x": "NAMED",
              "y": "NON_NULL_NUMERIC",
            },
          },
          Object {
            "points": Array [
              Object {
                "x": "Anything",
                "y": -50,
              },
              Object {
                "x": "Belongs",
                "y": 30,
              },
              Object {
                "x": "Conservatory",
                "y": 10,
              },
            ],
            "seriesName": "B",
            "type": Object {
              "x": "NAMED",
              "y": "NON_NULL_NUMERIC",
            },
          },
          Object {
            "points": Array [
              Object {
                "x": "Anything",
                "y": 50,
              },
              Object {
                "x": "Belongs",
                "y": -30,
              },
              Object {
                "x": "Conservatory",
                "y": 20,
              },
            ],
            "seriesName": "C",
            "type": Object {
              "x": "NAMED",
              "y": "NON_NULL_NUMERIC",
            },
          },
        ]
      }
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      title="My fourth bar chart"
      xAxisConfig={
        Object {
          "margin": 10,
          "tickLength": 2,
          "tickPlacement": 1,
          "zeroIntercept": true,
        }
      }
      yAxisConfig={
        Object {
          "margin": 10,
          "tickLength": 2,
          "tickPlacement": 0,
          "zeroIntercept": true,
        }
      }
    >
      Mock BarChart 
    </div>
  </div>
</div>
`;
